### ISSUE TITLE
Flashing content and CSS reliance

### ISSUE THREAD

**Issue by dd8** [w3c/wcag#2080]
It's possible to have flashing content that is usually hidden by CSS, or is usually displayed for too short a period to cause a failure of 2.3.1 / 2.3.2:

1) a flashing loader spinner where the load is usually under 1 sec (but sometimes runs for > 1 sec if network slow)
2) a flashing image that has `display:none` until the user selects No Style from the Firefox menu
3) a flashing image that has `display:none` until the user selects Safari Reader Mode
4) a flashing image that has `display:none` unless CSS load fails due to network conditions
5) a flashing image that has `display:none` unless certain CSS media feature conditions are met 
6) a flashing image that always visible - unless user.css specifies `img { display:none } `

This is related to the CSS reliance questions in https://github.com/w3c/wcag/issues/1554. 

The potential consequences for the user of all of these is the same - are only some WCAG failures?

**Comment by GreggVan** [comment]
I think that this case is a bit different.  In this case it is a health and safety risk - and he talks about CSS failing due to network failures — not someone messing with the CSS.

Maybe it is all moot since it is not clear why you would have flashing content that is not visible  - and this is just meant to test boundaries - or make some completely different point. 

I would suggest that we not spend much time debating this topic around with a flashing example unless there is some real question.  If there is no real place where this occurs or might occur and it is just meant to be a CSS discussion - then I suggest we choose another example.    To understand why I would suggest a CSS failure due to network delay - would mean (for most all other cases) that the CSS will arrive later or with a page refresh and the person can continue to use the site.   But a flash failure may leave them on the floor or on the way to the hospital.    Not the same thing.

Best

Gregg

———————————
Gregg Vanderheiden
***@***.***

**Comment by dd8** [comment]
> Maybe it is all moot since it is not clear why you would have flashing content that is not visible  - and this is just meant to test boundaries - or make some completely different point.

It's about which technologies you can rely on, and where in the page life cycle you test things:

- some SCs should be evaluated after everything has loaded and scripts have run - but this one shouldn't (IMO)
- some SCs can rely on CSS - but this one can't due to the risks (IMO) 

A likely path for all of the examples is a flashing loading spinner. Some frameworks include loading spinners in boilerplate HTML even if you don't show them - so an author may have unknowingly included a flickering image in their code, that's waiting to detonate given the right circumstances.

MDN specifically calls out loading spinners as potentially dangerous:
https://developer.mozilla.org/en-US/docs/Web/Accessibility/Seizure_disorders

> unless i'm misunderstanding, 6 would always fail flashing unless the user disables it. expecting the user to disable images is not a mechanism that can be relied on to say that the page/site passes.
Yes, that's in there as a counter-example. I'd expect that to always fail.

**Comment by fstrr** [comment]
This issue looks like it can be closed. If it needs to be re-opened, please start a new discussion.
